### PR TITLE
ddt removal in ui user tests

### PR DESCRIPTION
**Pull request by sthirugn**

### CODE REVIEW

**Comment by sthirugn** [comment]
test results:
```sh
# nosetests tests/foreman/ui/test_user.py:User.test_delete_user
.
----------------------------------------------------------------------
Ran 1 test in 77.123s

OK
# nosetests tests/foreman/ui/test_user.py:User.test_positive_create_user_different_usernames
.
----------------------------------------------------------------------
Ran 1 test in 86.106s

OK
# nosetests tests/foreman/ui/test_user.py:User.test_positive_create_user_different_first_names
.
----------------------------------------------------------------------
Ran 1 test in 94.673s

OK
# nosetests tests/foreman/ui/test_user.py:User.test_positive_create_user_different_surnames
.
----------------------------------------------------------------------
Ran 1 test in 94.729s

OK
# nosetests tests/foreman/ui/test_user.py:User.test_positive_create_user_different_languages
.
----------------------------------------------------------------------
Ran 1 test in 66.198s

OK
# nosetests tests/foreman/ui/test_user.py:User.test_positive_create_user_different_pass
.
----------------------------------------------------------------------
Ran 1 test in 107.574s

OK
# nosetests tests/foreman/ui/test_user.py:User.test_negative_create_user_with_invalid_name
.
----------------------------------------------------------------------
Ran 1 test in 44.079s

OK
# nosetests tests/foreman/ui/test_user.py:User.test_negative_create_user_with_invalid_firstname
.
----------------------------------------------------------------------
Ran 1 test in 36.702s

OK
# nosetests tests/foreman/ui/test_user.py:User.test_negative_create_user_with_invalid_surname
.
----------------------------------------------------------------------
Ran 1 test in 36.721s

OK
```

File: tests/foreman/ui/test_user.py [hunk 1]
```diff
@@ -249,11 +247,25 @@ def test_positive_create_user_different_pass(self, password):
         @Assert: User is created
 
         """
-        name = gen_string('alpha')
+        test_data = valid_strings()
```
Comment (sthirugn) [inline tests/foreman/ui/test_user.py]: Lines 250-257 can be very well combined in to the for statement in Line 259 but it makes the loop look so complex.  So I split it for readability.

**Comment by sthirugn** [comment]
Rebased

File: tests/foreman/ui/test_user.py [hunk 2]
```diff
@@ -203,18 +204,20 @@ def test_positive_create_user_different_languages(self, language):
...
+                    make_user(session, username=name, locale=language)
+                    element = self.user.search(name, search_key)
+                    self.assertIsNotNone(element)
+                    element.click()
+                    self.assertEqual(
+                        language,
+                        self.user.wait_until_element(
+                            locators['users.selected_lang']).get_attribute(
+                                'value')
+                        )
```
Comment (abalakh) [inline tests/foreman/ui/test_user.py]: nitpick:
```python
                    self.assertEqual(
                        language,
                        self.user.wait_until_element(
                            locators['users.selected_lang']
                        ).get_attribute('value')
                    )
```

**Comment by abalakh** [comment]
ACK pending small nitpick

File: tests/foreman/ui/test_user.py (hunk 2 above)
Comment (sthirugn) [inline tests/foreman/ui/test_user.py]: Sure :+1:

**Comment by sthirugn** [comment]
Comment addressed and rebased

File: tests/foreman/ui/test_user.py (hunk 1 above)
Comment (abalakh) [inline tests/foreman/ui/test_user.py]: If we used python3, things would be pretty simple:
```python
test_data = [
    u'foo@!#$^&*( ) {0}'.format(gen_string('alpha', 2)),
    u'bar+{{}}|\"?hi {0}'.format(gen_string('alpha', 2)),
    *valid_strings(),
]
```
However, AFAIK python2 doesn't support this syntax.

**Comment by omaciel** [comment]
ACK

**Comment by sthirugn** [comment]
Part of https://github.com/SatelliteQE/robottelo/issues/2769

**Pull request merged**